public-ui/src/app/components/db-no-results/no-results-search.component.tsx

```diff
@@ -0,0 +1,184 @@
+import { Configuration, DataBrowserApi } from 'publicGenerated/fetch';
+import * as React from 'react';
+import {navigate, navigateByUrl} from 'app/utils/navigation';
+import { reactStyles } from 'app/utils/index';
+
+const api = new DataBrowserApi(new Configuration({ basePath: environment.publicApiUrl }));
+
+const styles = reactStyles({
+    loadingText: {
+        fontFamily: 'GothamBook, Arial, sans-serif',
+        fontWeight: 'normal',
+        fontStyle: 'normal',
+        fontStretch: 'normal',
+        lineHeight: '1.5',
+        fontSize: '16px',
+        letterSpacing: 'normal',
+        textAlign: 'left',
+        color: '#262262',
+        paddingLeft: '0'
+  },
+  noResults: {
+    marginTop: '-1rem',
+    padding: '1em'
+  },
+  loadingDiv: {
+      display: 'flex',
+      flexDirection: 'row',
+      justifyContent: 'flex-start',
+      alignItems: 'center',
+      marginBottom: '1em'
+  },
+  spinnerDiv: {
+      marginTop: '1em',
+      marginBottom: '0.2em',
+      marginLeft: '0.5em'
+  }
+});
+
+const styleCss =
+`
+a:link,a:visited,a{
+    color:#2aa3d8;
+}
+`;
+
+interface Props {
+    searchValue: string;
+    measurementTestFilter: number;
+    measurementOrderFilter: number;
+    domainMatch: Function;
+}
+
+interface State {
+    domainInfoResults: Array<any>;
+    surveyModuleResults: Array<any>;
+    pmResults: Array<any>;
+    fitbitResults: Array<any>;
+    loading: boolean;
+}
+
+export const NoResultSearchComponent = (class extends React.Component<Props, State> {
+    constructor(props) {
+        super(props);
+        this.handleOnClick = this.handleOnClick.bind(this);
+        this.state = {
+            domainInfoResults: [],
+            surveyModuleResults: [],
+            pmResults: [],
+            fitbitResults: [],
+            loading: true
+        };
+    }
+
+    componentDidMount() {
+        this.fetchDomainTotals();
+    }
+
+    fetchDomainTotals() {
+        const {searchValue, measurementTestFilter, measurementOrderFilter} = this.props;
+        api.getDomainTotals(searchValue, measurementTestFilter, measurementOrderFilter).then(
+                result => {
+                    result.domainInfos = result.domainInfos.filter(domain =>
+                                        domain.standardConceptCount > 0);
+                    this.setState({domainInfoResults: result.domainInfos.filter(
+                    domain => domain.name.toLowerCase() !== 'physical measurements' &&
+                    domain.name.toLowerCase() !== 'fitbit'),
+                    surveyModuleResults: result.surveyModules,
+                    pmResults: result.domainInfos.filter(
+                    domain => domain.name.toLowerCase() === 'physical measurements'),
+                    fitbitResults: result.domainInfos.filter(
+                    domain => domain.name.toLowerCase() === 'fitbit')});
+                    this.setState({loading: false});
+                });
```

Comment:
Nit: consistent indentation, combine `setState`s
```suggestion
        api.getDomainTotals(searchValue, measurementTestFilter, measurementOrderFilter).then(result => {
            result.domainInfos = result.domainInfos.filter(domain => domain.standardConceptCount > 0);
            this.setState({
                domainInfoResults: result.domainInfos.filter(domain =>
                  domain.name.toLowerCase() !== 'physical measurements' && domain.name.toLowerCase() !== 'fitbit'),
                surveyModuleResults: result.surveyModules,
                pmResults: result.domainInfos.filter(domain => domain.name.toLowerCase() === 'physical measurements'),
                fitbitResults: result.domainInfos.filter(domain => domain.name.toLowerCase() === 'fitbit'),
                loading: false
            });
        });
```